### PR TITLE
Added support for collection multi level folders

### PR DESCRIPTION
**Pull request by kunagpal**
* Item names now show up as a full path, including names of parent folders

### CODE REVIEW

**Comment by kunagpal** [comment]
@shamasis https://github.com/postmanlabs/postman-runtime/pull/174

File: lib/reporters/cli/index.js [hunk 1]
```diff
@@ -45,14 +37,16 @@ colors.setTheme({
  * @param {Boolean=} reporterOptions.noFailures - Boolean flag to turn off failure reporting altogether, if set to true.
  * @param {Boolean=} reporterOptions.noConsole - Boolean flag to turn off console logging, if set to true.
  * @param {Object} options - A set of generic collection run options.
+ * @param {Object} nameCache - A set of item id - full name key pairs.
  * @returns {*}
  */
-PostmanCLIReporter = function (emitter, reporterOptions, options) {
+PostmanCLIReporter = function (emitter, reporterOptions, options, nameCache) {
```
Comment (czardoz) [inline lib/reporters/cli/index.js]: Please remove this cache thing.

Comment (kunagpal) [inline lib/reporters/cli/index.js]: Removed.

File: lib/reporters/html/partials/aggregations.hbs [hunk 2]
```diff
@@ -53,4 +51,8 @@
             </div>
         </div>
     </div>
+{{/each}}
+{{#if parent.name}}
```
Comment (czardoz) [inline lib/reporters/html/partials/aggregations.hbs]: This is not good, we should not have two related parts of the template in separate if blocks

File: lib/reporters/junit/index.js [hunk 3]
```diff
@@ -42,7 +44,7 @@ JunitReporter = function (newman, reporterOptions) {
 
             if (!currentItem) { return; }
 
-            suite.att('name', currentItem.name);
+            suite.att('name', util.getFullName(currentItem, options));
```
Comment (czardoz) [inline lib/reporters/junit/index.js]: Some utility function now has access to all the run options :/

File: lib/util.js [hunk 4]
```diff
@@ -54,6 +57,34 @@ module.exports = {
...
+     * @param {?String} [name=''] The intermediate name for the specified item
+     * @param {?String} [id=''] Placeholder id
+     * @returns {String} The full name of the provided item, including prepended parent item names
+     * @private
+     */
+    getFullName: function (item, options, name, id) {
+        var parent = this.parentOf(item); // @todo: Use item.parentOf when available
+
+        return parent ?
+            this.getFullName(parent, options, (item.name || item.id) + (name ? SEP + name : E), id ||
```
Comment (czardoz) [inline lib/util.js]: Don't use `this` here.

**Comment by shamasis** [comment]
I agree that util should not have all run options. That's bad coupling.

File: lib/reporters/html/partials/aggregations.hbs (hunk 2 above)
Comment (kunagpal) [inline lib/reporters/html/partials/aggregations.hbs]: Fixed.

File: lib/util.js (hunk 4 above)
Comment (kunagpal) [inline lib/util.js]: Fixed.

File: lib/reporters/junit/index.js (hunk 3 above)
Comment (kunagpal) [inline lib/reporters/junit/index.js]: Fixed.

File: lib/reporters/cli/index.js [hunk 5]
```diff
@@ -72,7 +62,7 @@ PostmanCLIReporter = function (emitter, reporterOptions, options) {
 
         // show the failures table (provided option does not say it is not to be shown)
         if (!reporterOptions.noFailures && run.failures && run.failures.length) {
-            print(LF + PostmanCLIReporter.parseFailures(run.failures).toString() + LF);
+            print(LF + PostmanCLIReporter.parseFailures(run.failures, options).toString() + LF);
```
Comment (kunagpal) [inline lib/reporters/cli/index.js]: This is from an older diff, and was removed on Friday 24 Feb 2017.

**Pull request merged**